code/HUXt.py

```diff
@@ -1576,3 +1732,241 @@ def _setup_dirs_():
+    
+    :param v_input: Time series of inner boundary solar wind speeds, in km/s.
+    :dt: The time step of v_input in seconds
+    
+    """
+    #convert input units
+    tmax=tmax.to('s')
+    rin=rin.to('km')
+    rout=rout.to('km')
+    dr=dr.to('km')
+    dtdr=dt/dr
+
+    #get the HUXt constants
+    constants= huxt_constants()
+    alpha = constants['alpha']  # Scale parameter for residual SW acceleration
+    r_accel = constants['r_accel']  # Spatial scale parameter for residual SW acceleration
+    #v_max = constants['v_max'] 
+    r_accel=r_accel.to('km')   
+    
+    #check the CFL condition is met
+    if (np.nanmax(v_input)>dr.to('km')/dt):
+        print('Warning: CFL condition not met, time step too short. Increase HUXt.constants.v_max')    
+
+    #get the radial grid parameters
+    Nr=int((rout-rin)/dr)
+    r= np.linspace(rin, rout, Nr)
+    rrel = r - r[0]
+    
+    voutindex=int(r.size)
+    n_steps=v_input.size
+    
+    # Initialise the arrays
+    vt = np.ones(Nr) * v_input[0]
+    urnext = np.ones(Nr) * np.nan *u.km/u.s
+    ur = np.ones(Nr) * np.nan *u.km/u.s
+    vsource=np.ones(Nr) * np.nan *u.km/u.s
+    vdiff=np.ones(Nr) * np.nan *u.km/u.s
+    times=np.ones(n_steps)*np.nan*u.s
+    speeds=np.ones(n_steps)*np.nan*u.km/u.s
+    times[0]=0.0
+    speeds[0]=v_input[0]   
+   
+    # loop through time and compute the updated 1-d radial solution
+    for t in range(1, n_steps):
+        #using the _upwind_step_opt_ function is slower by ~ x100
+        # # Pull out the upwind and downwind slices at current time
+        # u_up=vt[1:];  u_dn=vt[:-1]
+        # u_up_next = _upwind_step_opt_(u_up.to(u.km/u.s).value, 
+        #                                   u_dn.to(u.km/u.s).value,
+        #                                   dtdr.to(u.s/u.km).value, 
+        #                                   alpha, r_accel.to(u.km).value, rrel.to(u.km).value)
+        # # Save the updated time step
+        # vt[1:] = u_up_next.copy()*u.km/u.s
+        # #update the inner boundary value
+        # vt[0]=v_input[t]
+
+        #update each radial position
+        ur=vt
+        urnext[1:]=ur[1:] - dt*ur[1:]*(ur[1:]-ur[:-1])/dr 
+        #compute the source speed
+        vsource=ur[:-1]/(1 + alpha*(1 - np.exp(-rrel[:-1]/r_accel)) )
+        #compute the speed gain between r and r+dr
+        vdiff=alpha*vsource*(1-np.exp(-rrel[1:]/r_accel)) - alpha*vsource*(1-np.exp(-rrel[:-1]/r_accel))
+        #add the residual acceleration over this grid cell
+        urnext[1:]=urnext[1:] + (ur[:-1]*dt/dr)*vdiff
+        #update the grid
+        vt=urnext
+        #update the inner boundary value
+        vt[0]=v_input[t]
+        #save the output
+        times[t]=t*dt
+        speeds[t]=vt[voutindex-1]
+    return times,speeds
+        
+
+@jit(nopython=False)
+def solve_upwind_allR(v_input, dt=500*u.s, tmax=10*u.day, rin = 30.0 *u.solRad, rout= 215.0*u.solRad, dr=1.4747*u.solRad):
```

Comment:
I wonder if we can condense solve_upwind_Rout and solve_upwind_allR? For example, with a keyword argument like outer_only=False could return the speeds at all radial points and outer_only=True could just return the speeds at the specified radius? But this isn't really important.